geopandas/tests/test_op_output_types.py

```diff
@@ -235,3 +234,53 @@ def identity(x):
         return x
 
     assert_object(df[["geometry2"]].apply(identity, axis=1), GeoDataFrame, None, None)
+
+
+def test_expanddim_in_apply():
+    # https://github.com/geopandas/geopandas/pull/2296#issuecomment-1021966443
+    s = GeoSeries.from_xy([0, 1], [0, 1])
+    result = s.apply(lambda x: pd.Series([x.x, x.y]))
+    assert_object(result, pd.DataFrame)
+
+
+@pytest.mark.xfail(
+    not compat.PANDAS_GE_11,
+    reason="pandas <1.1 don't preserve subclass through groupby ops",  # Pandas GH33884
+)
+def test_expandim_in_groupby_aggregate_multiple_funcs():
+    # https://github.com/geopandas/geopandas/pull/2296#issuecomment-1021966443
+    # There are two calls to _constructor_expanddim here
+    # SeriesGroupBy._aggregate_multiple_funcs() and
+    # SeriesGroupBy._wrap_series_output() len(output) > 1
+
+    s = GeoSeries.from_xy([0, 1, 2], [0, 1, 3])
+
+    def union(s):
+        return s.unary_union
+
+    def total_area(s):
+        return s.area.sum()
+
+    grouped = s.groupby([0, 1, 0])
+    agg = grouped.agg([total_area, union])
+    assert_object(agg, GeoDataFrame, None, None, check_none_name=True)
+    result = grouped.agg([union, total_area])
+    assert_object(result, GeoDataFrame, None, None, check_none_name=True)
+    assert_object(grouped.agg([total_area, total_area]), pd.DataFrame)
+    assert_object(grouped.agg([total_area]), pd.DataFrame)
+
+
+@pytest.mark.xfail(
+    not compat.PANDAS_GE_11,
+    reason="pandas <1.1 uses concat([Series]) in unstack",  # Pandas GH33356
+)
+def test_expanddim_in_unstack():
+    # https://github.com/geopandas/geopandas/pull/2296#issuecomment-1021966443
+    s = GeoSeries.from_xy(
+        [0, 1, 2],
+        [0, 1, 3],
+        index=pd.MultiIndex.from_tuples([("A", "a"), ("A", "b"), ("B", "a")]),
+    )
+    unstack = s.unstack()
+    assert_object(unstack, GeoDataFrame, None, None, False)
+    assert unstack._geometry_column_name == "geometry"
```

Comment:
Following the code you added in `__finalize__` specifically for `method == "unstack"`, should this be equal to None?

Reply:
It seems this test is also failing in CI for that reason